apps/aecore/src/aec_db.erl

```diff
@@ -144,6 +148,7 @@ tables(Mode) ->
    , ?TAB(aec_tx_location)
    , ?TAB(aec_tx_pool)
    , ?TAB(aec_discovered_pof)
+   , ?TAB(aec_tx_gc)
```

Comment:
Call to `aec_db:add_tx/1` writes to table `aec_signed_tx` without deleting from `aec_tx_gc`. What prevents mempool push, namely [this call](https://github.com/aeternity/aeternity/blob/e9bd96779c585eab037727b008babce242dab2da/apps/aecore/src/aec_tx_pool.erl#L717) (please consider only atomic blocks), to write `aec_signed_tx` entry shortly after `aec_db:gc_tx/1`? This would lead to tx having entry in `aec_signed_tx` (and `aec_tx_pool`) and `aec_tx_gc`. (I may be missing something.) Is this expected? What is user (or state channel FSM?) experience in this case (e.g. event published anywhere for tx included in mempool)? And how does DB/mempool/GC behave in this case?

Reply:
`aec_db:add_tx/1` can be reach only after https://github.com/aeternity/aeternity/blob/master/apps/aecore/src/aec_tx_pool.erl#L169 returns `ok`. This is not transactional, but returning `ok` there means that there is no either `aec_tx_gc` or `aec_signed_tx` object for a transaction in the system. This means that there is no transaction in the system to be GCed (which prevents transaction from being GCed shortly).

Even is there is a path (as of now - I don't see any) which leads to having both `aec_signed_tx` and `aec_tx_gc` objects in the system - there will be no harm. New GC of transactions, namely `aec_tx_gc` server, removes only `aec_tx_gc` objects from the DB, without any impact on `aec_signed_tx`.